File: scripts/fix-all-linting.ts
#!/usr/bin/env tsx
/**
 * Comprehensive linting fix script
 * Fixes all common linting errors across the codebase
 */

import * as fs from 'fs';
import * as path from 'path';
import { glob } from 'glob';

interface Fix {
  pattern: RegExp;
  replacement: string | ((match: string, ...args: any[]) => string);
  description: string;
  files?: string[];
}

const fixes: Fix[] = [
  // Fix empty object patterns in function parameters
  {
    pattern: /export (?:async )?function (\w+)\(\s*\{\s*\}\s*:\s*(\w+)\s*\)/g,
    replacement: 'export async function $1(_args: $2)',
    description: 'Fix empty object patterns',
  },
  
  // Fix unused error variables
  {
    pattern: /const error = /g,
    replacement: 'const _error = ',
    description: 'Prefix unused error variables',
  },
  
  // Fix unused imports - specific cases
  {
    pattern: /import\s*\{\s*Link\s*,/g,
    replacement: (match, ...args) => {
      // Check if Link is used in the file
      return match; // Keep for now, needs context
    },
    description: 'Remove unused imports',
  },
  
  // Fix constant binary expressions
  {
    pattern: /if\s*\(\s*false\s*\|\|\s*/g,
    replacement: 'if (',
    description: 'Remove false || from conditions',
  },
  {
    pattern: /\s*\|\|\s*false\s*\)/g,
    replacement: ')',
    description: 'Remove || false from conditions',
  },
  {
    pattern: /if\s*\(\s*true\s*&&\s*/g,
    replacement: 'if (',
    description: 'Remove true && from conditions',
  },
  
  // Fix specific unused variables
  {
    pattern: /const\s+(filter|formId|scheduleId|searchParams|setSearchParams|totalCount|metrics|recentCases|upcomingTasks|clients|accounts|filters|rateTables|mfaRequired|Milestone|RiskFactor|caseId|matterType|reason)\s*=/g,
    replacement: 'const _$1 =',
    description: 'Prefix specific unused variables',
  },
  
  // Fix unused function parameters
  {
    pattern: /\(data\)\s*=>/g,
    replacement: '(_data) =>',
    description: 'Prefix unused data parameter',
  },
  {
    pattern: /\(loaderData\)\s*=>/g,
    replacement: '(_loaderData) =>',
    description: 'Prefix unused loaderData parameter',
  },
  {
    pattern: /function\s+\w+\s*\(\s*request\s*:/g,
    replacement: (match) => match.replace('request', '_request'),
    description: 'Prefix unused request parameter',
  },
];

async function fixFile(filePath: string): Promise<{ fixed: boolean; changes: string[] }> {
  let content = fs.readFileSync(filePath, 'utf8');
  const changes: string[] = [];
  let fixed = false;

  for (const fix of fixes) {
    const before = content;
    
    if (typeof fix.replacement === 'function') {
      content = content.replace(fix.pattern, fix.replacement);
    } else {
      content = content.replace(fix.pattern, fix.replacement);
    }
    
    if (content !== before) {
      changes.push(fix.description);
      fixed = true;
    }
  }

  if (fixed) {
    fs.writeFileSync(filePath, content, 'utf8');
  }

  return { fixed, changes };
}

async function main() {
  console.log('🔧 Starting comprehensive linting fixes...\n');

  // Get all TypeScript/TSX files
  const files = await glob('frontend/src/**/*.{ts,tsx}', {
    ignore: ['**/node_modules/**', '**/*.d.ts', '**/dist/**'],
  });

  console.log(`Found ${files.length} files to process\n`);

  let totalFixed = 0;
  const fileResults: Array<{ file: string; changes: string[] }> = [];

  for (const file of files) {
    const result = await fixFile(file);
    
    if (result.fixed) {
      totalFixed++;
      fileResults.push({ file, changes: result.changes });
      console.log(`✓ Fixed: ${file}`);
      result.changes.forEach(change => console.log(`  - ${change}`));
    }
  }

  console.log(`\n📊 Summary:`);
  console.log(`  Total files processed: ${files.length}`);
  console.log(`  Files fixed: ${totalFixed}`);
  console.log(`\n✅ Done! Run 'cd frontend && npm run lint' to verify.`);
}

main().catch(console.error);

File: scripts/fix-all-linting.py
#!/usr/bin/env python3
"""
Comprehensive linting error fixer for the LexiFlow frontend
Handles:
- Unused variables (prefix with _)
- Empty object patterns  
- Unused error variables in catch blocks
- no-case-declarations
- no-useless-catch
- no-constant-binary-expression
"""

import re
import os
from pathlib import Path

# Define all fixes
FIXES = [
    # Fix unused error variables in catch blocks
    {
        'files': [
            'frontend/src/features/operations/billing/fee-agreements/FeeAgreementManagement.tsx',
            'frontend/src/features/operations/billing/rate-tables/RateTableManagement.tsx',
            'frontend/src/features/litigation/evidence/EvidenceForensics.tsx',
            'frontend/src/services/data/repositories/DocumentRepository.tsx',
            'frontend/src/services/domain/AdminDomain.ts',
            'frontend/src/services/domain/SearchDomain.ts',
            'frontend/src/services/domain/WarRoomDomain.ts',
            'frontend/src/services/integration/backendDiscovery.ts',
            'frontend/src/hooks/useImageOptimization.ts',
        ],
        'pattern': r'} catch \(error\) \{',
        'replace': '} catch (_error) {',
    },
    # Fix unused function parameters
    {
        'files': [
            'frontend/src/features/cases/components/docket/DocketSheet.tsx',
        ],
        'pattern': r'const \[startTransition\]',
        'replace': 'const [_startTransition]',
    },
    {
        'files': [
            'frontend/src/features/cases/components/list/CaseManagement.tsx',
        ],
        'pattern': r'\(onSelectCase\)',
        'replace': '(_onSelectCase)',
    },
    {
        'files': [
            'frontend/src/routes/cases/create.tsx',
        ],
        'pattern': r'\(loaderData\)',
        'replace': '(_loaderData)',
    },
    {
        'files': [
            'frontend/src/routes/documents/index.tsx',
        ],
        'pattern': r'const \[totalCount,',
        'replace': 'const [_totalCount,',
    },
    {
        'files': [
            'frontend/src/routes/billing/rates.tsx',
        ],
        'pattern': r'const \[rateTables,',
        'replace': 'const [_rateTables,',
    },
    {
        'files': [
            'frontend/src/features/dashboard/components/EnhancedDashboardOverview.tsx',
        ],
        'pattern': r'\buserRole\b(?=\s*[:=])',
        'replace': '_userRole',
    },
    {
        'files': [
            'frontend/src/features/dashboard/components/role-dashboards/AdminDashboard.tsx',
            'frontend/src/features/dashboard/components/role-dashboards/AttorneyDashboard.tsx',
            'frontend/src/features/dashboard/components/role-dashboards/ParalegalDashboard.tsx',
        ],
        'pattern': r'const \[roleDashboard\]',
        'replace': 'const [_roleDashboard]',
    },
    {
        'files': [
            'frontend/src/features/dashboard/components/role-dashboards/PartnerDashboard.tsx',
        ],
        'pattern': r'const \[roleDashboard\]',
        'replace': 'const [_roleDashboard]',
    },
    {
        'files': [
            'frontend/src/features/dashboard/components/role-dashboards/PartnerDashboard.tsx',
        ],
        'pattern': r'const \[billingData\]',
        'replace': 'const [_billingData]',
    },
    {
        'files': [
            'frontend/src/features/litigation/discovery/DiscoveryTimeline.tsx',
        ],
        'pattern': r', index\)',
        'replace': ', _index)',
    },
    {
        'files': [
            'frontend/src/features/litigation/discovery/LegalHoldsEnhanced.tsx',
        ],
        'pattern': r'interface LegalHoldNotification',
        'replace': '// interface LegalHoldNotification',
    },
    {
        'files': [
            'frontend/src/features/operations/documents/DocumentExplorer.tsx',
        ],
        'pattern': r', index\)',
        'replace': ', _index)',
    },
    {
        'files': [
            'frontend/src/features/litigation/strategy/AICommandBar.tsx',
        ],
        'pattern': r'\b_e\b(?=\s*[,)])',
        'replace': '__e',
    },
    {
        'files': [
            'frontend/src/hooks/useBreadcrumbs.ts',
        ],
        'pattern': r', index\)',
        'replace': ', _index)',
    },
    {
        'files': [
            'frontend/src/utils/billing-features.ts',
        ],
        'pattern': r'\(reason\)',
        'replace': '(_reason)',
    },
    {
        'files': [
            'frontend/src/utils/route-guards.ts',
        ],
        'pattern': r'\(request\)',
        'replace': '(_request)',
    },
    {
        'files': [
            'frontend/src/services/infrastructure/apiClientEnhanced.ts',
        ],
        'pattern': r'class ApiError',
        'replace': '// class ApiError',
    },
    # Fix empty object patterns
    {
        'files': [
            'frontend/src/routes/admin/audit.tsx',
        ],
        'pattern': r'export default function \w+\(\{\}\)',
        'replace': 'export default function AuditPage()',
    },
    {
        'files': [
            'frontend/src/routes/admin/backup.tsx',
        ],
        'pattern': r'export default function \w+\(\{\}\)',
        'replace': 'export default function BackupPage()',
    },
    {
        'files': [
            'frontend/src/routes/admin/integrations.tsx',
        ],
        'pattern': r'export default function \w+\(\{\}\)',
        'replace': 'export default function IntegrationsPage()',
    },
    {
        'files': [
            'frontend/src/routes/admin/settings.tsx',
        ],
        'pattern': r'export default function \w+\(\{\}\)',
        'replace': 'export default function SettingsPage()',
    },
    {
        'files': [
            'frontend/src/routes/dashboard.tsx',
        ],
        'pattern': r'export default function \w+\(\{\}\)',
        'replace': 'export default function DashboardPage()',
    },
    {
        'files': [
            'frontend/src/routes/layout.tsx',
        ],
        'pattern': r'export default function \w+\(\{\}\)',
        'replace': 'export default function Layout()',
    },
]

def apply_fix(file_path, pattern, replace):
    """Apply a single fix to a file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if isinstance(pattern, str):
            pattern = re.compile(pattern)
        
        new_content = pattern.sub(replace, content)
        
        if new_content != content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            return True
        return False
    except Exception as e:
        print(f"  ✗ Error: {e}")
        return False

def main():
    """Apply all fixes"""
    root = Path.cwd()
    fixed_count = 0
    error_count = 0
    
    print("Applying linting fixes...\n")
    
    for fix in FIXES:
        pattern = fix['pattern']
        replace = fix['replace']
        
        for file_rel in fix['files']:
            file_path = root / file_rel
            
            if not file_path.exists():
                print(f"⚠️  File not found: {file_rel}")
                continue
            
            if apply_fix(file_path, pattern, replace):
                print(f"✓ {file_rel}")
                fixed_count += 1
    
    print(f"\n✓ Applied fixes to {fixed_count} files")
    if error_count > 0:
        print(f"✗ {error_count} errors encountered")

if __name__ == '__main__':
    main()